tor/core/admin_commands.py

```diff
@@ -89,17 +167,14 @@ def process_blacklist(reply, config):
             failed.append(username)
             continue
 
-        already_added = config.redis.sadd('blacklist', username)
-        if already_added == 0:
+        if config.redis.sadd('blacklist', username):
```

Comment:
I think you might mean `if not config.redis.sadd('blacklist', username):` here. We're trying to detect if adding to the set fails, yes? If it cannot (and returns zero), this method would return a falsey value.